lib/wickedbase.pm

```diff
@@ -559,8 +559,11 @@ sub post_run {
     $self->do_barrier('post_run');
     if ($self->{name} ne 'before_test' && get_var('WICKED_TCPDUMP')) {
         script_run('kill ' . get_var('WICKED_TCPDUMP_PID'));
-        select_console('root-virtio-terminal1') if (get_var('VIRTIO_CONSOLE_NUM', 1) > 1);
-        upload_file('/tmp/tcpdump' . $self->{name} . '.pcap', 'tcpdump' . $self->{name} . '.pcap');
+        my $result = eval {
```

Comment:
I know we did it before. But I think it's not the best way to handle `eval{}` blocks.
Eval returns also the last statement return value or just what will be returned like a normal function.
So if upload_file() return `undef` or `0`, for what ever reason, we will announce a error where no is.
e.g.
```
perl -e 'my $r = eval { print "FOO$/"; undef; }; die("Ooops") unless($r);'
```
I think handling the error like this is the safer way.
```
perl -e 'my $r = eval { print "FOO$/"; undef; }; die("Ooops") if($@);'
```

Reply:
1. not sure that it will work , I remember I have seen runs where `$@` was empty for unknown reason . So can you be sure that $@ **always** contains something ? 
2. currently I am checking for `unless defined $result` so don't know why you mention `0` here ? We should speak only for `undef`. 
3. Code of upload_file is under our control  so in case we would face such scenario we could just make sure that upload_file not returning `undef`

Reply:
>     1. not sure that it will work , I remember I have seen runs where `$@` was empty for unknown reason . So can you be sure that $@ **always** contains something ?
 not sure if it is `undef` or just empty string  but still https://openqa.suse.de/tests/3247134#step/before_test/1

Reply:
https://openqa.suse.de/tests/3247134#step/before_test/1 me neither.. And also don't know the code anymore.

1. It's documented here. https://perldoc.perl.org/functions/eval.html => empty string on no error.
2. You are right
3. For me it isn't a valid point. Currently upload_file() just return the last statement value which is a Boolean. But when someone add a other call after it, because what ever, he might miss to realize that it change the return value and we get failures in a "hidden" way.

Reply:
done

Reply:
minor thing, I think we do not need `$result` anymore.

Reply:
done